From f43bbdf8a1b5d19f8c9b1ca45a4a14eff616edd8 Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:17:14 -0400
Subject: [net] netfilter: nf_tables: fix wrong format in request_module()

Message-id: <20141021161714.22888.45935.stgit@localhost.localdomain>
Patchwork-id: 97932
O-Subject: [RHEL7 net PATCH 035/124] netfilter: nf_tables: fix wrong format in request_module()
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: 2fec6bb6f484b1a88b4a325724234d6cfd08c918

 commit 2fec6bb6f484b1a88b4a325724234d6cfd08c918
 Author: Pablo Neira Ayuso <pablo@netfilter.org>
 Date:   Mon Mar 31 12:26:39 2014 +0200

    netfilter: nf_tables: fix wrong format in request_module()

    The intended format in request_module is %.*s instead of %*.s.

    Reported-by: Florian Westphal <fw@strlen.de>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netfilter/nf_tables_api.c b/net/netfilter/nf_tables_api.c
index 3ff4a80..85a7b12 100644
--- a/net/netfilter/nf_tables_api.c
+++ b/net/netfilter/nf_tables_api.c
@@ -152,8 +152,8 @@ nf_tables_chain_type_lookup(const struct nft_af_info *afi,
 #ifdef CONFIG_MODULES
  if (autoload) {
   nfnl_unlock(NFNL_SUBSYS_NFTABLES);
-  request_module("nft-chain-%u-%*.s", afi->family,
-          nla_len(nla)-1, (const char *)nla_data(nla));
+  request_module("nft-chain-%u-%.*s", afi->family,
+          nla_len(nla), (const char *)nla_data(nla));
   nfnl_lock(NFNL_SUBSYS_NFTABLES);
   type = __nf_tables_chain_type_lookup(afi->family, nla);
   if (type != NULL)
-- 
1.7.1